src/reconcile/rec_row.c

```diff
@@ -1190,6 +1190,10 @@ __wti_rec_row_leaf(
 
         /* Boundary: split or write the page. */
         if (__wt_rec_need_split(r, key->len + val->len)) {
+            /* We cannot split a page that is restored from deltas. */
```

Comment:
Why is that?

Reply:
In page read, we cannot handle page split. We expect the page to rewrite to a single page.

Reply:
Alright, let me see if I understand this:
* We can now reconcile from the read path, if the page has deltas
* The read path can't split *any* page, not just one with deltas

If that's correct, I have two follow-on questions:
1. Do we need to be careful that we never set the `WT_REC_REWRITE_DELTA` flag on the write path? Otherwise we'll always give up on split.
2. Is it feasible to just write the page even though it needs a split? If we're here due to reconstructing deltas, then presumably the page required a split even when it was read. So it seems like reconciling it in that state is not doing any harm.

Reply:
Spoke about this offline -- I didn't realise how urgent this was. There's already a TODO for improving things, so I'll leave this conversation open and approve.